test_suite/conftest.py

```diff
@@ -7,6 +7,28 @@
 from lib import test_lib
 
 
+@pytest.fixture(autouse=True)
+def check_run_on(host, request):
+    # This fixture checks the list of combinations of distro-version
+    # each test has. If the host doesn't meet any of the combinations,
+    # the test will be skipped
+    fedora_versions = ['f34', 'f35', 'f36']
+    rhel_versions = ['rhel8.5', 'rhel8.6', 'rhel9.0']
+    centos_versions = ['cs8', 'cs9']
+
+    distro_version_list = [m.args for m in request.node.own_markers if m.name == 'run_on'][0][0]
+
+    host_distro = host.system_info.distribution
+    host_version = host.system_info.release
+
+    if host_distro not in distro_version_list:
+        if (host_distro == 'fedora' and f'f{host_version}' not in fedora_versions) \
+            or (host_distro == 'rhel' and f'rhel{host_version}' not in rhel_versions) \
+                or (host_distro == 'centos' and f'cs{host_version}' not in centos_versions):
+
+            pytest.skip(f'This test does not apply to {host_distro} - {host_version}')
+
+
 @pytest.fixture
 def rhel_only(host):
```

Comment:
yes

Reply:
Ok, let's be careful since there may be other fixtures that rely on that one. E.g. `rhel_sap_only`. But I think it's easy to fix, we just need to add the new marker to `rhel_sap_only`.